filecoindot-rpc/src/lib.rs

```diff
@@ -47,15 +47,21 @@ impl<T> FilecoindotApi for Filecoindot<T>
 where
     T: OffchainStorage + 'static,
 {
-    fn set_rpc_endpoint(&self, url: String) -> Result<()> {
-        if url.starts_with("http") {
-            self.storage
-                .write()
-                .set(sp_offchain::STORAGE_PREFIX, FILECOIN_RPC, &url.encode());
-            Ok(())
-        } else {
-            Err(Error::InvalidEndpoint)
+    fn set_rpc_endpoint(&self, urls: String) -> Result<()> {
```

Comment:
Maybe use the [url crate](https://crates.io/crates/url) for a more accurate checking?